src/repository/views.py

```diff
@@ -355,8 +355,11 @@ def repository_search(request, search_term=None):
     )
 
     if search_term:
+        search_term = search_term.strip()
         split_search_term = search_term.split(' ')
```

Comment:
you'll probably need to clean the terms themselves as well, think of an accidental double space in a search term: it would lead to the icontains getting again an empty string to match upon. e.g `"Martin␣Paul␣␣Eve"`